messaging/Controllers/BundlesController.cs

```diff
@@ -104,35 +117,149 @@ public async Task<ActionResult<IncomingMessageItem>> GetIncomingMessageItem(stri
+                        responseBundle.Entry.Add(respEntry);
+                    }
+                    return responseBundle;
+                } 
+                else
+                {
+
+                    IncomingMessageItem item;
+                    try
+                    {
+                        item = ParseIncomingMessageItem(jurisdictionId, text);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogDebug($"An exception occurred while parsing the incoming message: {ex}");
+                        return BadRequest("Failed to parse message. Please verify that it is consistent with the current Vital Records Messaging FHIR Implementation Guide.");
+                    }
+
+                    // Pre-check some minimal requirements for validity. Specifically, if there are problems with the message that will lead to failure when
+                    // attempting to insert into the database (e.g. missing MessageId), catch that here to return a 400 instead of a 500 on DB error
+                    // Message errors SHOULD result in an ExtractionError response; this check is just to catch things that can't make it that far
+                    if (item.MessageId == null)
+                    {
+                        _logger.LogDebug("Rejecting message with no MessageId");
+                        return BadRequest("Message was missing required field MessageId");
+                    }
+                    if (item.MessageType == null)
+                    {
+                        _logger.LogDebug("Rejecting message with no MessageType.");
+                        return BadRequest("Message was missing required field MessageType");
+                    }
+
+                    item.Source = GetMessageSource();
+
+                    try
+                    {
+                        await SaveIncomingMessageItem(item, queue);
+                    }
+                    catch (Exception ex)
+                    {
+                        _logger.LogDebug($"An exception occurred while saving the incoming message: {ex}");
+                        return StatusCode(500);
+                    }
+                }
+
+                // return HTTP status code 204 (No Content)
+                return NoContent();
+            } catch (Exception ex){
+                Console.WriteLine($"An exception occurred while parsing the incoming message: {ex}");
+                return BadRequest();
+            }
+        }
+
+        // InsertBatchMessages handles a single message in a batch upload submission
+        // Each message is handled independent of the other messages. A status code is generated for
+        // each message and is returned in the response bundle
+        private async Task<Bundle.EntryComponent> InsertBatchMessages( Bundle.EntryComponent msgBundle, string jurisdictionId, [FromServices] IBackgroundTaskQueue queue)

```

Comment:
The `FromServices` attribute shouldn't be needed here, since we are explicitly passing in an instance rather than relying on the framework to inject the dependency.

Reply:
removed from this function and the ParseIncomingMessageItem function